addons/weapononback/functions/fnc_swap.sqf

```diff
@@ -0,0 +1,85 @@
+#include "script_component.hpp"
+/*
+ * Author: BaerMitUmlaut
+ * Swaps primary and weapon on back. Can handle no primary or no weapon on back.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+
+private _oldPrimary = (getUnitLoadout ACE_player)#0;
+private _newPrimary = [ACE_player] call FUNC(get);
+
+// Save scope adjustment
+private _adjustments = ACE_player getVariable [QEGVAR(scopes,adjustment), [[0, 0, 0], [0, 0, 0], [0, 0, 0]]];
+private _oldAdjustment = _adjustments#0;
+private _newAdjustement = ACE_player getVariable [QGVAR(scopeAdjustment), [0, 0, 0]];
+EGVAR(scopes,guns) set [0, ""];
+
+ACE_player removeWeapon (_oldPrimary param [0, ""]);
+
+if !(_newPrimary isEqualTo []) then {
+    _newPrimary params ["_weapon"];
+
+    // addWeapon is going to eat a mag for each muzzle before we can fill the weapon with addWeaponItem
```

Comment:
You need to check if weapon was empty and allow "eating" magazines.
Because currently swapping empty weapon will allow duping magazines.